src/test/java/bio/terra/service/filedata/azure/blobstore/AzureBlobStorePdaoTest.java

```diff
@@ -0,0 +1,278 @@
+  }
+
+  @Test
+  public void testDeleteFile() {
+    UUID fileId = UUID.randomUUID();
+    FSFileInfo fsFileInfo = mockFileCopy(fileId);
+    doNothing().when(blobCrl).deleteBlob(fileId + "/" + SOURCE_FILE_NAME);
+
+    FireStoreFile fileToDelete =
+        new FireStoreFile()
+            .fileId(fileId.toString())
+            .bucketResourceId(RESOURCE_ID.toString())
+            .gspath(fsFileInfo.getGspath());
+    assertThat(dao.deleteFile(fileToDelete), equalTo(true));
+  }
+
+  @Test
+  public void testDeleteFileNotFound() {
+    UUID fileId = UUID.randomUUID();
+    FSFileInfo fsFileInfo = mockFileCopy(fileId);
+    BlobStorageException exception = mock(BlobStorageException.class);
+    when(exception.getStatusCode()).thenReturn(HttpStatus.NOT_FOUND.value());
+    doThrow(exception).when(blobCrl).deleteBlob(fileId + "/" + SOURCE_FILE_NAME);
+
+    FireStoreFile fileToDelete =
+        new FireStoreFile()
+            .fileId(fileId.toString())
+            .bucketResourceId(RESOURCE_ID.toString())
+            .gspath(fsFileInfo.getGspath());
+    assertThat(dao.deleteFile(fileToDelete), equalTo(false));
+  }
+
+  @Test
+  public void testDeleteFileMismatchedStorageAccount() {
+    UUID fileId = UUID.randomUUID();
+    mockFileCopy(fileId);
+
+    FireStoreFile fileToDelete =
+        new FireStoreFile()
+            .fileId(fileId.toString())
+            .bucketResourceId(RESOURCE_ID.toString())
+            .gspath("https://differentaccountname.blob.core.windows.net/data/blob.txt");
+    assertThat(
+        assertThrows(PdaoException.class, () -> dao.deleteFile(fileToDelete)).getMessage(),
+        equalTo(
+            "Resource groups between metadata storage and request do not match: "
+                + "differentaccountname != sa"));
+  }
+
+  @Test
+  public void testDeleteFileById() {
+    UUID fileId = UUID.randomUUID();
+    mockFileCopy(fileId);
+    doNothing().when(blobCrl).deleteBlob(fileId + "/" + SOURCE_FILE_NAME);
+
+    assertThat(
+        dao.deleteDataFileById(fileId.toString(), SOURCE_FILE_NAME, AZURE_STORAGE_ACCOUNT_RESOURCE),
+        equalTo(true));
+  }
+
+  @Test
+  public void testDeleteFileByIdNotFound() {
+    UUID fileId = UUID.randomUUID();
+    mockFileCopy(fileId);
+    BlobStorageException exception = mock(BlobStorageException.class);
+    when(exception.getStatusCode()).thenReturn(HttpStatus.NOT_FOUND.value());
+    doThrow(exception).when(blobCrl).deleteBlob(fileId + "/" + SOURCE_FILE_NAME);
+
+    assertThat(
+        dao.deleteDataFileById(fileId.toString(), SOURCE_FILE_NAME, AZURE_STORAGE_ACCOUNT_RESOURCE),
+        equalTo(false));
+  }
+
+  @Test
+  public void testSasValidation() {
+    assertThat(
+        "is valid",
+        AzureBlobStorePdao.isSignedUrl(
+            "https://src.blob.core.windows.net/srcdata/src.txt"
+                + "?sp=r&st=2021-07-14T19:31:16Z&se=2021-07-15T03:31:16Z&spr=https&sv=2020-08-04&"
+                + "sr=b&sig=mysig"),
+        equalTo(true));
```

Comment:
There is no value is using a matcher to compare against `true` or `false. Instead of writing `assertThat(..., equalTo(true))` you can write
```suggestion
    assertThat(
        "is valid",
        AzureBlobStorePdao.isSignedUrl(
            "https://src.blob.core.windows.net/srcdata/src.txt"
                + "?sp=r&st=2021-07-14T19:31:16Z&se=2021-07-15T03:31:16Z&spr=https&sv=2020-08-04&"
                + "sr=b&sig=mysig"));
```
For `false` you can use `assertFalse()`.

Reply:
Oh cool, it does that by default?

Reply:
Oh wait assertTrue

Reply:
`MatcherAssert.assertThat(String, boolean)` is the same as `Assert.assertTrue(String, boolean)`. There is no analog in `MatcherAssert` for `assertFalse()`. Either `assertThat()` or `assertTrue()` is fine of course; I do kind of like `assertThat` better.

Reply:
I guess I like the symetry of assertTrue and assertFalse

Reply:
Sure, either is better than `assertThat(x, is(true))`